### ISSUE TITLE
API Changed?

### ISSUE THREAD

**Issue by folofjc** [zackdn/wf-readwise-integration#6]
First off, thanks for this! I have been using it and have loved having my readwise in WF!

Recently, however, it hasn't been working. I opened the console in chrome and am getting errors when I try to run the bookmarklet:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'highlights')
    at <anonymous>:1:10864
    at Array.forEach (<anonymous>)
    at $jscomp.generator.Engine_.program_ (<anonymous>:1:10831)
    at $jscomp.generator.Engine_.nextStep_ (<anonymous>:1:8133)
    at $jscomp.generator.Engine_.next_ (<anonymous>:1:7070)
    at $jscomp.generator.Generator_.next (<anonymous>:1:8581)
    at b (<anonymous>:1:9010)
(anonymous) @ VM183:1
(anonymous) @ VM183:1
$jscomp.generator.Engine_.nextStep_ @ VM183:1
$jscomp.generator.Engine_.next_ @ VM183:1
next @ VM183:1
b @ VM183:1
document_view.js?v=local:2 
        
        
       POST https://sentry.workflowy.com/api/3/envelope/?sentry_key=XXX&sentry_version=7&sentry_client=sentry.javascript.browser%2F7.70.0 net::ERR_BLOCKED_BY_CLIENT
(anonymous) @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
add @ document_view.js?v=local:2
f @ document_view.js?v=local:2
_sendEnvelope @ document_view.js?v=local:2
sendSession @ document_view.js?v=local:2
captureSession @ document_view.js?v=local:2
_updateSessionFromEvent @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
_executeHandlers @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
u @ document_view.js?v=local:2
then @ document_view.js?v=local:2
_processEvent @ document_view.js?v=local:2
_captureEvent @ document_view.js?v=local:2
captureEvent @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
_withClient @ document_view.js?v=local:2
captureEvent @ document_view.js?v=local:2
m @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
v @ document_view.js?v=local:2
l.onunhandledrejection @ document_view.js?v=local:2
document_view.js?v=local:2 
        
        
       POST https://sentry.workflowy.com/api/3/envelope/?sentry_key=XXX&sentry_version=7&sentry_client=sentry.javascript.browser%2F7.70.0 net::ERR_BLOCKED_BY_CLIENT
(anonymous) @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
add @ document_view.js?v=local:2
f @ document_view.js?v=local:2
_sendEnvelope @ document_view.js?v=local:2
sendEvent @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
_executeHandlers @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
u @ document_view.js?v=local:2
then @ document_view.js?v=local:2
_processEvent @ document_view.js?v=local:2
_captureEvent @ document_view.js?v=local:2
captureEvent @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
_withClient @ document_view.js?v=local:2
captureEvent @ document_view.js?v=local:2
m @ document_view.js?v=local:2
(anonymous) @ document_view.js?v=local:2
v @ document_view.js?v=local:2
l.onunhandledrejection @ document_view.js?v=local:2
```

Did the API of WF or Readwise change?

**Comment by folofjc** [comment]
It is dying in the function `getAll HighlightsByBook` in the line `c[e.book_id].highlights.unshift(e)`

**Comment by folofjc** [comment]
Okay, so I think it was either an issue with one of my highlights, or an update problem. I kept changing the date in the node that gave my last sync. I just moved it one day into the future at a time and it eventually updated. I am guess that I lost some data but I went through my nodes and couldn't find it. I will close this, but some better error handling that told me what the problem was would be nice!
